bindings/node/lib/clientEncryption.js

```diff
@@ -299,6 +302,14 @@ module.exports = function (modules) {
 
         contextOptions.keyAltName = bson.serialize({ keyAltName });
       }
+      if (options.algorithm === 'Indexed') {
+        delete contextOptions.algorithm;
+        contextOptions.indexType = 'Equality';
+      }
+      if (options.algorithm === 'Unindexed') {
+        delete contextOptions.algorithm;
+        contextOptions.indexType = 'None';
+      }
```

Comment:
Let me know if I missed it somewhere but the spec states `contentionFactor` must default to 0 and also an error must be raised if it is set and the algorithm is not `'Indexed'`. I don't think I see that in this PR but let me know if I'm wrong.

Reply:
The spec is being descriptive here, not prescriptive – i.e. these things are handled and tested by libmongocrypt. It would actually be an error to explicitly set a default value for `contentionFactor` in the `'Unindexed'` case, even if that value matches the default.

Reply:
Thanks for the explanation.